### DISCUSSION TITLE
Configuring Frigate to store on network share

### DISCUSSION THREAD

**Discussion by jastden** [blakeblackshear/frigate#14920] [Ask A Question]
Hello all,
when adding a network folder to my docker-compose in the portainer, I get an error:
Failed to deploy a stack: time="2024-11-11T11:13:59Z" level=warning msg="/data/compose/1/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion" Network frigate_default Creating Network frigate_default Created Container frigate Creating Container frigate Created Container frigate Starting Error response from daemon: error while mounting volume '/var/lib/docker/volumes/frigate_samba-volume/_data': failed to mount local volume: mount //10.10.10.45/Storage:/var/lib/docker/volumes/frigate_samba-volume/_data, data: username=*******,password=********,vers=3.0: operation not permitted
My docker-compose:
```dockerfile
version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable-tensorrt
    deploy:    # <------------- Add this section
      resources:
        reservations:
           devices:
             - driver: nvidia
               count: 1 # number of GPUs
               capabilities: [gpu]
    environment:
      - YOLO_MODELS=yolov7-320
      - USE_FP16=false
      - HA_TOKEN=*******
    shm_size: "600mb" # update for your cameras based on calculation above
    devices:
      - /dev/bus/usb:/dev/bus/usb  # Passes the USB Coral, needs to be modified for other versions
      - /dev/apex_0:/dev/apex_0    # Passes a PCIe Coral, follow driver instructions here https://coral.ai/docs/m2/get-started/#2a-on-linux
      - /dev/video11:/dev/video11  # For Raspberry Pi 4B
      - /dev/dri/renderD128:/dev/dri/renderD128 # For intel hwaccel, needs to be updated for your hardware
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /opt/frigate/config:/config
#      - /opt/frigate/storage:/media/frigate
      - samba-volume:/media/frigate
      - /opt/frigate/config:/db/
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "1984:1984"
      - "9150:5000"
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
volumes:
  samba-volume:
    driver_opts:
      type: "cifs"
      o: "username=******,password=******,vers=3.0"
      device: "//10.10.10.45/Storage"
``` 
My network folder is located in Synology at: //10.10.10.45/Storage
I will be grateful for your help!!!

**Comment by dimatx** [comment]
I'd recommend mounting your network share on your OS (outside of docker) under /mnt/your-share-here, and then point docker to /mnt/your-share-here on the host side. Also, if your Docker host and Synology support NFS, that is generally going to be more performant.

**Reply by jastden** [reply]
> I'd recommend mounting your network share on your OS (outside of docker) under /mnt/your-share-here, and then point docker to /mnt/your-share-here on the host side. Also, if your Docker host and Synology support NFS, that is generally going to be more performant.

Дмитрий, это все и есть одна машина. У меня есть сервер, на нем стоит Proxmox, в нем стоит  VM HAOS, LXC frigate, VM Synology. В сервере установлен SSD и два HDD, первый HDD подключен к LXC frigate, второй к VM Synology, на SSD стоит VM HAOS. Задумка в том что бы отдать оба HDD в Synology, а уже из него пробросить во Frigate, для того что бы иметь доступ ко всем записям и файлам из synology. Это же возможно, просто не очень понимаю как. Буду благодарен за любую помощь!

**Reply by bubba925** [reply]
You would use something like autofs to mount the NFS share in your Frigate LXC from the synology and then use that mounted path in the docker compose.

**Reply by jastden** [reply]
Unfortunately, I can't mount the NFS storage, because it's impossible to mount the NFS in an unpriviliged LXC container :(  Any other thoughts on this?

**Reply by dimatx** [reply]
My approach is to mount the nfs share on the _host_, so if you are running Proxmox, on Proxmox host. That way, your LXC simply needs to map to a directory on your Proxmox host. I use option 3 in this post: https://forum.proxmox.com/threads/tutorial-mounting-nfs-share-to-an-unprivileged-lxc.138506/
